src/main/java/scmspain/karyon/restrouter/RestRouterHandler.java

```diff
@@ -0,0 +1,222 @@
+package scmspain.karyon.restrouter;
+
+import com.google.common.net.HttpHeaders;
+import com.google.common.net.MediaType;
+import io.netty.buffer.ByteBuf;
+import io.reactivex.netty.protocol.http.server.HttpServerRequest;
+import io.reactivex.netty.protocol.http.server.HttpServerResponse;
+import io.reactivex.netty.protocol.http.server.RequestHandler;
+import org.apache.commons.lang.StringUtils;
+import org.commonjava.mimeparse.MIMEParse;
+import rx.Observable;
+import scmspain.karyon.restrouter.exception.CannotSerializeException;
+import scmspain.karyon.restrouter.exception.InvalidAcceptHeaderException;
+import scmspain.karyon.restrouter.handlers.DefaultErrorHandler;
+import scmspain.karyon.restrouter.handlers.ErrorHandler;
+import scmspain.karyon.restrouter.handlers.RestRouterErrorDTO;
+import scmspain.karyon.restrouter.serializer.SerializeManager;
+import scmspain.karyon.restrouter.serializer.SerializeWriter;
+import scmspain.karyon.restrouter.serializer.Serializer;
+import scmspain.karyon.restrouter.transport.http.RestUriRouter;
+import scmspain.karyon.restrouter.transport.http.Route;
+import scmspain.karyon.restrouter.transport.http.RouteNotFound;
+
+import javax.inject.Inject;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class RestRouterHandler implements RequestHandler<ByteBuf, ByteBuf> {
```

Comment:
You are right about the move, the problem is that we divide this class in two: RestRouterHandler and RestRouterScanner to be able to do unit testing of this parts.
While we done this division we just cut/paste code and delete the original class when we finish all the job.

I'm agree about that this PR is too much. It was quite complicated to start to do unit testing of this accept part with the current abstraction. About the number of files, some files are just imports reorganized, this shouldn't be part of this PR, sorry about that.

If it is necessary I can rework those commits and squash it.